data/json/effects_on_condition/addictions_eocs.json

```diff
@@ -135,69 +72,29 @@
-                ]
-              },
-              { "u_add_morale": "morale_craving_mutagen", "bonus": -20, "max_bonus": -200 }
-            ]
-          },
+        "if": { "math": [ "rand(600)", "<=", "(600 / (600 - 50 * u_addiction_intensity('mutagen'))) - 1" ] },
+        "then": {
+          "if": { "math": [ "rand(6)", "<", "u_addiction_intensity('mutagen')" ] },
+          "then": { "u_message": "You so miss the exquisite rainbow of post-humanity.", "type": "warning" },
+          "else": { "u_message": "Your body is SOO booorrrring.  Just a little sip to liven things up?", "type": "warning" }
+        },
+        "else": { "u_add_morale": "morale_craving_mutagen", "bonus": -20, "max_bonus": -200 }
+      },
+      {
+        "if": {
+          "and": [
+            { "math": [ "u_val('focus')", ">", "40" ] },
+            { "math": [ "rand(800)", "<=", "(800 / (800 - 20 * u_addiction_intensity('mutagen'))) - 1" ] }
+          ]
+        },
+        "then": [
+          { "math": [ "u_val('focus')", "-=", "u_addiction_intensity('mutagen')" ] },
           {
-            "id": "EOC_MUTAGEN_ADDICTION_JUNKIE_FOCUS",
-            "condition": {
-              "and": [
-                { "compare_num": [ { "u_val": "focus" }, ">", { "const": 40 } ] },
-                {
-                  "compare_num": [ { "rand": 800 }, "<=", { "u_val": "addiction_intensity", "addiction": "mutagen", "mod": { "val": 800, "step": 20 } } ]
-                }
-              ]
-            },
-            "effect": [
-              { "arithmetic": [ { "u_val": "focus" }, "-=", { "u_val": "addiction_intensity", "addiction": "mutagen" } ] },
-              {
-                "u_message": "You daydream what it'd be like if you were *different*.  Different is good.",
-                "type": "warning"
-              }
-            ]
+            "u_message": "You daydream what it'd be like if you were *different*.  Different is good.",
+            "type": "warning"
           }
         ]
       }
-    ],
-    "false_effect": {
-      "run_eocs": [
-        {
-          "id": "EOC_MUTAGEN_ADDICTION_NOJUNKIE",
-          "condition": {
-            "compare_num": [ { "rand": 500 }, "<=", { "u_val": "addiction_intensity", "addiction": "mutagen", "mod": { "val": 500, "step": 15 } } ]
-          },
-          "effect": [
-            {
-              "run_eocs": [
-                {
-                  "id": "EOC_MUTAGEN_ADDICTION_NOJUNKIE_MSG",
-                  "condition": { "compare_num": [ { "rand": 6 }, "<", { "u_val": "addiction_intensity", "addiction": "mutagen" } ] },
-                  "effect": { "u_message": "You haven't had any mutagen lately.", "type": "warning" },
-                  "false_effect": { "u_message": "You could use some new parts…", "type": "warning" }
-                }
-              ]
-            },
-            { "u_add_morale": "morale_craving_mutagen", "bonus": -5, "max_bonus": -50 }
-          ]
-        }
-      ]
-    }
+    ]
```

Comment:
This one's easier in a patch

<details>
<summary>Note that there's a max() in there that needs to go in the jmath function</summary>

```diff
diff --git a/data/json/effects_on_condition/addictions_eocs.json b/data/json/effects_on_condition/addictions_eocs.json
index b864d42267..e12e6606ea 100644
--- a/data/json/effects_on_condition/addictions_eocs.json
+++ b/data/json/effects_on_condition/addictions_eocs.json
@@ -72,13 +72,15 @@
     "condition": { "u_has_trait": "MUT_JUNKIE" },
     "effect": [
       {
-        "if": { "math": [ "rand(600)", "<=", "(600 / (600 - 50 * u_addiction_intensity('mutagen'))) - 1" ] },
-        "then": {
-          "if": { "math": [ "rand(6)", "<", "u_addiction_intensity('mutagen')" ] },
-          "then": { "u_message": "You so miss the exquisite rainbow of post-humanity.", "type": "warning" },
-          "else": { "u_message": "Your body is SOO booorrrring.  Just a little sip to liven things up?", "type": "warning" }
-        },
-        "else": { "u_add_morale": "morale_craving_mutagen", "bonus": -20, "max_bonus": -200 }
+        "if": { "math": [ "rand(600)", "<=", "(600 / max(1, (600 - 50 * u_addiction_intensity('mutagen')))) - 1" ] },
+        "then": [
+          {
+            "if": { "math": [ "rand(6)", "<", "u_addiction_intensity('mutagen')" ] },
+            "then": { "u_message": "You so miss the exquisite rainbow of post-humanity.", "type": "warning" },
+            "else": { "u_message": "Your body is SOO booorrrring.  Just a little sip to liven things up?", "type": "warning" }
+          },
+          { "u_add_morale": "morale_craving_mutagen", "bonus": -20, "max_bonus": -200 }
+        ]
       },
       {
         "if": {
@@ -95,6 +97,19 @@
           }
         ]
       }
+    ],
+    "false_effect": [
+      {
+        "if": { "math": [ "rand(500)", "<=", "(500 / (500 - 15 * u_addiction_intensity('mutagen'))) - 1" ] },
+        "then": [
+          {
+            "if": { "math": [ "rand(6)", "<", "u_addiction_intensity('mutagen')" ] },
+            "then": { "u_message": "You haven't had any mutagen lately.", "type": "warning" },
+            "else": { "u_message": "You could use some new parts…", "type": "warning" }
+          },
+          { "u_add_morale": "morale_craving_mutagen", "bonus": -5, "max_bonus": -50 }
+        ]
+      }
     ]
   }
 ]
```

</details>